src/scripts/items/EvolutionStone.ts

```diff
@@ -14,33 +14,40 @@ class EvolutionStone extends Item {
         player.gainItem(GameConstants.StoneType[this.type], n)
     }
 
-
-
     public use(pokemon?:string) {
         let shiny = PokemonFactory.generateShiny(GameConstants.SHINY_CHANCE_STONE);
-        if(pokemon == "Eevee") {
-            switch (this.type) {
+        let evolution = EvolutionStone.computeEvolution(this.type, pokemon);
+        player.capturePokemon(evolution, shiny, false);
```

Comment:
Or build the check here, see other comment

Reply:
I don't think we ever run into that problem, because if a stone is coded incorrectly, then `PokemonHelper.getPokemonsWithEvolution` would see that a pokemon doesn't evolve with this stone, and wouldn't show this pokemon on the UI. So the player has no way of selecting it. 

Another way of saying this is, players are only able to select the pokemons that we have coded correctly. So an incorrect coding won't result in page rendering failure; it'll result in the base form pokemon not showing.

Reply:
Good point